tools/msys2_fetch_and_build_all_tx16s.sh

```diff
@@ -15,81 +41,109 @@ do
 	fi
 done
 
+STEP=1
 echo "=== Step $((STEP++)): Creating a directory for EdgeTX ==="
-mkdir ~/edgetx && cd ~/edgetx
+mkdir -p ~/edgetx && cd ~/edgetx
+check_command $? "mkdir -p ~/edgetx && cd ~/edgetx"
 if [[ $PAUSEAFTEREACHLINE == "true" ]]; then
   echo "Step finished. Please check the output above and press Enter to continue or Ctrl+C to stop."
   read
 fi
 
-echo "=== Step $((STEP++)): Fetching EdgeTX source tree (main branch) from GitHub ==="
-git clone --recursive -b main https://github.com/EdgeTX/edgetx.git edgetx_main
+
+if [ ! -d "${SOURCE_DIR}" ]; then
+  echo "=== Step $((STEP++)): Fetching EdgeTX source tree (${BRANCH_NAME} branch) from GitHub ==="
+  git clone --recursive -b ${BRANCH_NAME} https://github.com/EdgeTX/edgetx.git ${SOURCE_DIR}
+  check_command $? "git clone --recursive -b ${BRANCH_NAME} https://github.com/EdgeTX/edgetx.git ${SOURCE_DIR}"
+  cd ${SOURCE_DIR}
+else
+  echo "=== Step $((STEP++)): updating EdgeTX source tree (${BRANCH_NAME} branch) from GitHub ==="
+  cd ${SOURCE_DIR}
+  check_command $? "cd ${SOURCE_DIR}"
+  git checkout ${BRANCH_NAME}
+  check_command $? "git checkout ${BRANCH_NAME}"
+  git pull
+  check_command $? "git pull"
+fi
 if [[ $PAUSEAFTEREACHLINE == "true" ]]; then
   echo "Step finished. Please check the output above and press Enter to continue or Ctrl+C to stop."
   read
 fi
 
 echo "=== Step $((STEP++)): Creating build output directory ==="
-cd edgetx_main && mkdir build-output && cd build-output
+mkdir -p build-output && cd build-output
+check_command $? "mkdir -p build-output && cd build-output"
 if [[ $PAUSEAFTEREACHLINE == "true" ]]; then
   echo "Step finished. Please check the output above and press Enter to continue or Ctrl+C to stop."
   read
 fi
 
-echo "=== Step $((STEP++)): Running CMake for TX16S as an example ==="
-cmake -G "MSYS Makefiles" -Wno-dev -DCMAKE_PREFIX_PATH=$HOME/5.12.9/mingw73_64 -DSDL_LIBRARY_PATH=/mingw64/bin/ -DPCB=X10 -DPCBREV=TX16S -DDEFAULT_MODE=2 -DGVARS=YES -DPPM_UNIT=US -DHELI=NO -DLUA=YES -DCMAKE_BUILD_TYPE=Release ../
+echo "=== Step $((STEP++)): Running CMake for ${RADIO_TYPE} as an example ==="
+cmake -G "MSYS Makefiles" -Wno-dev -DCMAKE_PREFIX_PATH=$HOME/5.12.9/mingw73_64 -DSDL_LIBRARY_PATH=/mingw64/bin/ -DPCB=X10 -DPCBREV=${RADIO_TYPE} -DDEFAULT_MODE=2 -DGVARS=YES -DPPM_UNIT=US -DHELI=NO -DLUA=YES -DCMAKE_BUILD_TYPE=Release ../
```

Comment:
If you want to be able to give radio type via variable, then also PCB should be dynamically changeable. Please note that some radio targets only use PCB and do NOT use PCBREV.

Reply:
not familiar with the actual build
can you point me to the list of options?

Reply:
See the build-<manufacturer>.py scripts for hints under https://github.com/EdgeTX/edgetx/tree/main/tools